### PR TITLE
Charms

### PR DESCRIPTION
**Pull request by ArturKnopik**
### Pull Request Prelude

- [x] I have followed [proper The Forgotten Server code styling][code].
- [x] I have read and understood the [contribution guidelines][cont] before making this PR.
- [x] I am aware that this PR may be closed if the above-mentioned criteria are not fulfilled.

### Changes Proposed

Introduction of the charm system(offensive perks + dodge) 
image

https://github.com/otland/forgottenserver/issues/3871 issue

### CODE REVIEW

File: data/lib/core/player.lua [hunk 1]
```diff
@@ -737,3 +742,37 @@ function Player.disableLoginMusic(self)
 	msg:delete()
 	return true
 end
+
+function Player.isCharmUnlocked(self, charmId)
+	if self:getStorageValue(PlayerStorageKeys.charmsUnlocked + charmId) == 1 then
```
Comment (MillhioreBT) [inline data/lib/core/player.lua:747]: Storages now return nil if empty, doing a direct check with a number is wrong

Comment (ArturKnopik) [reply data/lib/core/player.lua:747]: that is why this PR is on draft, i porting it from my personal project to TFS

File: data/lib/core/player.lua [hunk 2]
```diff
@@ -523,6 +523,11 @@ function Player.addBestiaryKills(self, raceId)
 			break
 		end
 	end
+
+	if kills < bestiaryInfo.mastery and newKills >= bestiaryInfo.mastery then
+		self:addCharmPoints(bestiaryInfo.charmPoints)
+	end
```
Comment (ranisalt) [inline data/lib/core/player.lua:529]: Can this be added to `unlockCharm` (and `removeCharm`) instead?

File: data/lib/core/player.lua [hunk 3]
```diff
@@ -737,3 +742,51 @@ function Player.disableLoginMusic(self)
...
+	UNLOCKED = 1,
+	LOCKED = 0,
+}
+
+function Player.isCharmUnlocked(self, charmId)
+	if (self:getStorageValue(PlayerStorageKeys.charmsUnlocked + charmId) or 0) == charmStatus.UNLOCKED then
+		return charmStatus.UNLOCKED
+	end
+	return charmStatus.LOCKED
+end
```
Comment (ranisalt) [inline data/lib/core/player.lua:756]: Would

```suggestion
function Player.isCharmUnlocked(self, charmId)
	return self:getStorageValue(PlayerStorageKeys.charmsUnlocked + charmId) or 0) or charmStatus.LOCKED
end
```

work?

**Pull request closed without merging**